src/CreateNewContactFormPage.jsx

```diff
@@ -0,0 +1,169 @@
+import PropTypes from "prop-types";
+import { useState } from "react";
+import "./App.css";
+import { useNavigate } from "react-router-dom";
+function CreateNewContactFormPage(props) {
+  const { setConstactInfoList, contactInfoList } = props;
+  const [newContact, setNewContact] = useState({
+    firstName: "",
+    lastName: "",
+    street: "",
+    city: "",
+    email: "",
+    favouriteColour: "",
+    profileImage: "",
+  });
+
+  const navigate = useNavigate();
+
+  function handelInputData(event) {
+    const inputName = event.target.name;
+    const inputValue = event.target.value;
+
+    if (inputName === "firstName") {
+      setNewContact({ ...newContact, firstName: inputValue });
+    }
+    if (inputName === "lastName") {
+      setNewContact({ ...newContact, lastName: inputValue });
+    }
+    if (inputName === "email") {
+      setNewContact({ ...newContact, email: inputValue });
+    }
+    if (inputName === "street") {
+      setNewContact({ ...newContact, street: inputValue });
+    }
+    if (inputName === "city") {
+      setNewContact({ ...newContact, city: inputValue });
+    }
+    if (inputName === "colour") {
+      setNewContact({ ...newContact, favouriteColour: inputValue });
+    }
+    if (inputName === "url") {
+      setNewContact({ ...newContact, profileImage: inputValue });
+    }
+  }
+  function handleSubmit() {
+    console.log(newContact);
```

Comment:
don't forget: event.preventDefault(); to prevent the default form submission action